wdc-calculator-core/wdc-calculator-core-action/src/main/java/hu/icellmobilsoft/wdc/calculator/core/action/WorkdayCache.java

```diff
@@ -48,14 +49,34 @@
 
     private Set<Year> guaranteedYears = new HashSet<>();
 
+    private final Object lockObject = new Object();
+
     /**
      * Returns the cache of workdays
      *
      * @return Map<Year, TreeMap<LocalDate, WorkdayCacheData>> Map with initialized years as keys and TreeMaps as values. Every TreeMap has LocalDates as keys
      *         (every date in the given year) and a WorkdayCacheData as value.
      */
     public Map<Year, TreeMap<LocalDate, WorkdayCacheData>> getCache() {
-        return cache;
+        synchronized (lockObject) {
```

Comment:
`synchronized (this) {` ?

Reply:
We could, but will still need a getter for it, because of bean proxy.